src/lang/modifyAst.ts

```diff
@@ -224,8 +245,21 @@ export function mutateKwArg(
   for (let i = 0; i < node.arguments.length; i++) {
     const arg = node.arguments[i]
     if (arg.label.name === label) {
-      node.arguments[i].arg = val
-      return true
+      if (isLiteralArrayOrStatic(val) && isLiteralArrayOrStatic(arg.arg)) {
+        node.arguments[i].arg = val
+        return true
+      } else if (
+        arg.arg.type === 'ArrayExpression' &&
+        val.type === 'ArrayExpression'
+      ) {
+        const arrExp = arg.arg
+        arrExp.elements.forEach((element, i) => {
+          if (isLiteralArrayOrStatic(element)) {
+            arrExp.elements[i] = val.elements[i]
+          }
+        })
+        return true
+      }
```

Comment:
This fixes an issue where draging a `line` segment handle would nuke constraints.

Reply:
This is that issue you mentioned to me that got introduced with kwargs yeah? Nice catch

Reply:
Yeah sorry should have mentioned that, specific to kwarg segments, which pretty much means `line` atm.